eva-accession-ws/src/main/java/uk/ac/ebi/eva/accession/ws/rest/SubmittedVariantsRestController.java

```diff
@@ -93,5 +107,60 @@ public SubmittedVariantsRestController(
                         service.getLastInactive(identifier),
                         SubmittedVariant::new));
     }
-}
 
+    @GetMapping(value = "/beacon/query", produces = "application/json")
+    public BeaconAlleleResponse doesVariantExist(@RequestParam(name="assemblyId") String assembly,
+                                                 @RequestParam(name="referenceName") String chromosome,
+                                                 @RequestParam(name="study") String study,
+                                                 @RequestParam(name="start") long start,
+                                                 @RequestParam(name="referenceBases") String reference,
+                                                 @RequestParam(name="alternateBases") String alternate,
+                                                 HttpServletResponse response) {
+        if (start < 1) {
+            int responseStatus = HttpServletResponse.SC_BAD_REQUEST;
+            response.setStatus(responseStatus);
+            return getBeaconResponseObjectWithError(alternate, reference, chromosome, start, assembly,
+                                                    responseStatus, "Please provide a positive number as start position");
+        }
+        try {
+            return beaconService.queryBeacon(null, alternate, reference, chromosome, start, assembly, false, study);
+        }
+        catch (Exception ex) {
+            int responseStatus = HttpServletResponse.SC_INTERNAL_SERVER_ERROR;
+            response.setStatus(responseStatus);
+            return getBeaconResponseObjectWithError(alternate, reference, chromosome, start, assembly,
+                                                    responseStatus, "Unexpected Error: " + ex.getMessage());
+        }
+    }
+
+    private BeaconAlleleResponse getBeaconResponseObjectWithError(String alternate, String reference, String chromosome,
+                                                                  long start, String assembly, int errorCode,
+                                                                  String errorMessage) {
+        BeaconAlleleResponse result = new BeaconAlleleResponse();
+        BeaconAlleleRequest request = new BeaconAlleleRequest(alternate, reference, chromosome, start,
+                                                              assembly, null, false);
+        result.setAlleleRequest(request);
+        result.setError(new BeaconError(errorCode, errorMessage));
+        return result;
+    }
+
+    @ApiOperation(value = "Find submitted variants (SS) by the identifying fields", notes = "This endpoint returns the submitted "
+            + "variants (SS) represented by a given identifier. For a description of the response, see "
+            + "https://github.com/EBIvariation/eva-accession/wiki/Import-accessions-from-dbSNP#submitted-variant-subsnp-or-ss")
+    @GetMapping(value = "/by-id-fields", produces = "application/json")
```

Comment:
this is also unintuitive for me. I would call this "/by-fields" or even leaving just "/v1/submitted-variants", although I don't know if the latter is possible.

Reply:
@cyenyxe @jmmut How does "by-locus" sound?

Reply:
Sidestepped this getting rid of the endpoint altogether